lib/matplotlib/colorbar.py

```diff
@@ -424,7 +424,6 @@ def __init__(self, ax, mappable=None, *, cmap=None,
         for spine in self.ax.spines.values():
             spine.set_visible(False)
         self.outline = self.ax.spines['outline'] = _ColorbarSpine(self.ax)
-        self._short_axis().set_visible(False)
```

Comment:
I'll be surprised if this works - I think you will still have ticks showing up on the short axis

Reply:
Isn't the baseline image ok?
Checking some trivial examples it seemed fine but I had a feeling I'm breaking something and I couldn't figure it out!

Reply:
It may be worth checking some other formats (ps, eps, svg), at least manually. It does indeed look like it works, no doubt about that, but quite surprising...

Reply:
I checked some examples in eps, svg and pdf and they were the same. It doesn't seem that the output format has any effects on it.

Reply:
I believe it works because `_draw_all` calls `self._short_axis().set_ticks([])`.